pkg/util/kubernetes/kubelet/pod.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubelet
+
+package kubelet
+
+import (
+	"encoding/json"
+
+	log "github.com/cihub/seelog"
+)
+
+type creatorRef struct {
+	Kind      string
+	Reference PodOwner
+}
+
+// Owners returns pod owners, sourced either from:
+// - the new `Owners` field, exposed by the kubelet since 1.6
+// - the legacy `kubernetes.io/created-by` annotation, deprecated in 1.8
+func (p *Pod) Owners() []PodOwner {
+	// If we find the new field, return it
+	owners := p.Metadata.Owners
+	if len(owners) > 0 {
+		return owners
+	}
+
+	// Else, try unserialising the legacy field
+	content, found := p.Metadata.Annotations["kubernetes.io/created-by"]
+	if !found {
+		return owners
```

Comment:
As len(owners) == 0, why don't we return nil or something that would state that the owner was not found more explicitely?